lib/components/viewers/route-viewer.js

```diff
@@ -33,14 +33,42 @@ class RouteViewer extends Component {
   static propTypes = {
     hideBackButton: PropTypes.bool,
     routes: PropTypes.object
-  }
+  };
+
+  state = {
+    agency: null,
+    mode: null,
+    search: ''
+  };
```

Comment:
Slight pref for this stuff to live in redux, and leverage a selector to get the list of routes (rather than doing the filtering within the render method).

Reply:
Moved to redux in e886cfc75dde3dbec1844daa3cd88c51a0750e23